main.js

```diff
@@ -1 +1,61 @@
+class Calculator {
+  constructor(previousOperandTextElement, currentOperandTextElement) {
+    this.previousOperandTextElement = previousOperandTextElement
+    this.currentOperandTextElement = currentOperandTextElement
+    this.clear()
+  }
 
+  clear() {
+    this.currentOperand = ''
+    this.previousOperand = ''
+    this.operation = undefined
+  }
+
+  delete() {
+
+  }
+
+  appendNumber(number) {
+    if (number === '.' && this.currentOperand.includes('.')) return
+    this.currentOperand = this.currentOperand.toString() + number.toString()
+  }
+
+  chooseOperation(operation) {
+    this.operation = operation
+    this.previousOperand = this.currentOperand
+    this.currentOperand = ''
+  }
+
+  compute() {
+
+  }
+
+  updateDisplay() {
+    this.currentOperandTextElement.innerText = this.currentOperand
+  }
+}
+
+
+const numberButtons = document.querySelectorAll('[data-number]')
+const operationButtons = document.querySelectorAll('[data-operation]')
+const equalsButton = document.querySelector('[data-equals')
+const deleteButton = document.querySelector('[data-delete')
+const allClearButton = document.querySelector('[data-all-clear')
+const previousOperandTextElement = document.querySelector('[data-previous-operand')
+const currentOperandTextElement = document.querySelector('[data-current-operand')
+
+const calculator = new Calculator(previousOperandTextElement, currentOperandTextElement)
+
+numberButtons.forEach(buttons => {
+  button.addEventListener('click', () => {
+    calculator.appendNumber(button.innerText)
+    calculator.updateDisplay()
+  })
+})
+
+operationButtons.forEach(button => {
+  button.addEventListener('click', () => {
+    calculator.chooseOperation(button.innerText)
+    calculator.updateDisplay
+  })
+})
```

Comment:
Nice! Love to see the class constructors for object manipulation. This is clean code. Curious about some of the syntax, particularly on lines 41-46. Should there be a closing bracket on those data attributes or is it working just fine? Either way, good job stepping out your comfort zone to tackle this!